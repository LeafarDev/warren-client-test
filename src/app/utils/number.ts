export function toFixed(num: number, fixed: number): number {
  const re = new RegExp('^-?\\d+(?:\.\\d{0,' + (fixed || -1) + '})?');
  const numMatch = num.toString()
    .match(re);
  if (numMatch === undefined || numMatch === null) {
    return 0;
  }
  if (numMatch.length) {
    return Number(numMatch[0]);
  }
  return 0;
}
