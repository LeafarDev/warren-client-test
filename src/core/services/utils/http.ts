import axios from 'axios';

const getToken = () => {
  let token = null;

  if (localStorage.getItem('persist:bankofmars')) {
    // @ts-ignore
    const { auth } = JSON.parse(localStorage.getItem('persist:bankofmars'));
    token = JSON.parse(auth).token;
  }
  return token || 'OAuthAccessToken';
};

export default axios.create({
  baseURL: process.env.REACT_APP_API,
  headers: {
    Authorization: getToken()
  }
});
