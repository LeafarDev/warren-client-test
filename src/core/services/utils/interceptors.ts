import api from './http';
import { store } from '../../store/configureStore';
import { logout } from '../../actionCreators/authActionCreators';

const onExpireToken = () => {
  api.interceptors.response.use(
    (response) => response,
    (error) => {
      const status = error.response ? error.response.status : error.status;
      const errorMessage = error.message || '';
      const unauthorizedPattern = /status code 401/i;

      if ((status === 401 || errorMessage.match(unauthorizedPattern)) &&
        error.response.data.message !== 'Username or password incorrect') {
        store.dispatch(logout());
        return Promise.reject();
      }

      return Promise.reject(error);
    }
  );
};

export default {
  onExpireToken
};
