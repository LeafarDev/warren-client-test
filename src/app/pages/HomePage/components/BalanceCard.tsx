import React from 'react';
import { BalanceGrid } from '../styles';
import { Box, Card, CardContent, Grid, Typography } from '@material-ui/core';
import currency from 'currency.js';
import { toFixed } from '../../../utils/number';

interface Props {
  amount: number;
  title: string;
}

export const BalanceCard: React.FC<Props> = props => {
  const {
    amount,
    title
  } = props;

  return (
    <BalanceGrid item xs={12} sm={6} md={3}>
      <Grid>
        <Grid item xs={12} sm={12}>
          <Card variant="outlined">
            <CardContent>
              <Box display={'flex'} alignContent={'center'} alignItems={'center'}
                   flexDirection={'column'}>
                <Typography color="textSecondary" gutterBottom>
                  {title}
                </Typography>
                <Typography variant="h4" component="h2">
                  {`${currency(toFixed(amount, 2), {
                    decimal: '.',
                    precision: 2
                  })
                    .format()}`}
                </Typography>
              </Box>
            </CardContent>
          </Card>
        </Grid>
      </Grid>
    </BalanceGrid>
  );
};
