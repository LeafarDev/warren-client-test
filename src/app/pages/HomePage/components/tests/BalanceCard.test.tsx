import React from 'react';
import { BalanceCard } from '../BalanceCard';
import { render, screen } from '@testing-library/react';
import currency from 'currency.js';
import { toFixed } from '../../../../utils/number';

describe('<BalanceCard />', () => {
  const amount = 500;
  const title = 'Testing BalanceCard';

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild }
    } = render(
      <BalanceCard amount={amount} title={title}>
      </BalanceCard>);

    expect(firstChild)
      .toMatchSnapshot();
  });

  it('should render a formatted amount', () => {
    render(<BalanceCard amount={amount} title={title}></BalanceCard>);

    const expectedFormatedTest = currency(toFixed(amount, 2), {
      decimal: '.',
      precision: 2
    })
      .format();
    expect(screen.getByText(expectedFormatedTest))
      .toBeInTheDocument();
  });
});
