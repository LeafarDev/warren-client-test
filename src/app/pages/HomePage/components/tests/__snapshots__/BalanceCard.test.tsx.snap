// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BalanceCard /> should render and match the snapshot 1`] = `
<div
  class="MuiGrid-root sc-bdnxRM jjVypx MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
>
  <div
    class="MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-outlined MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <div
            class="MuiBox-root MuiBox-root-1"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
            >
              Testing BalanceCard
            </p>
            <h2
              class="MuiTypography-root MuiTypography-h4"
            >
              $500.00
            </h2>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
